source/PreferencesPanel.cpp

```diff
@@ -251,14 +284,19 @@ bool PreferencesPanel::Click(int x, int y, int clicks)
 {
 	EndEditing();
 
-	if(x >= 265 && x < 295 && y >= -220 && y < 70)
+	Point point(x, y);
+	const Interface *preferencesUI = GameData::Interfaces().Get("preferences");
+	if(preferencesUI->GetBox("volume box").Contains(point))
 	{
-		Audio::SetVolume((17 - y) / 200.);
-		Audio::Play(Audio::Get("warder"));
+		Rectangle box = preferencesUI->GetBox("volume box");
+		double barSize = preferencesUI->GetValue("master volume bar size");
+		double volume = (box.Center().Y() - point.Y()) / barSize + 0.5;
```

Comment:
(of course github won't let me do it in one suggestion because deleted lines)
```suggestion
		double barSize = preferencesUI->GetValue("master volume bar size");
		double volume = (volumeBox.Center().Y() - point.Y()) / barSize + .5;
```